google/cloud/internal/rest_response.cc

```diff
@@ -0,0 +1,212 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "google/cloud/internal/rest_response.h"
+#include "google/cloud/internal/absl_str_cat_quiet.h"
+#include "google/cloud/log.h"
+#include <nlohmann/json.hpp>
+#include <iostream>
+
+namespace google {
+namespace cloud {
+namespace rest_internal {
+GOOGLE_CLOUD_CPP_INLINE_NAMESPACE_BEGIN
+namespace {
+// Maps HTTP status codes to enumerators in `StatusCode`. The code here is
+// organized by increasing range (or value) of the errors, just to make it
+// readable. There are probably shorter (and/or more efficient) ways to write
+// this, but we went for readability.
+StatusCode MapHttpCodeToStatus(long code) {  // NOLINT(google-runtime-int)
```

Comment:
Most of the code in this function is not tested.  That is Okay if you are planning to add the tests as part of the next PR